lib/workers/branch/execute-post-upgrade-commands.ts

```diff
@@ -0,0 +1,197 @@
+        } else {
+          logger.warn(
+            {
+              cmd,
+              allowedPostUpgradeCommands,
+            },
+            'Post-upgrade task did not match any on allowed list'
+          );
+          artifactErrors.push({
+            lockFile: upgrade.packageFile,
+            stderr: sanitize(
+              `Post-upgrade command '${cmd}' does not match allowed pattern${
+                allowedPostUpgradeCommands.length === 1 ? '' : 's'
+              } ${allowedPostUpgradeCommands.map((x) => `'${x}'`).join(', ')}`
+            ),
+          });
+        }
+      }
+
+      const status = await getRepoStatus();
+
+      for (const relativePath of status.modified.concat(status.not_added)) {
+        for (const pattern of fileFilters) {
+          if (minimatch(relativePath, pattern)) {
+            logger.debug(
+              { file: relativePath, pattern },
+              'Post-upgrade file saved'
+            );
+            const existingContent = await readLocalFile(relativePath);
+            const existingUpdatedArtifacts = updatedArtifacts.find(
+              (ua) => ua.name === relativePath
+            );
+            if (existingUpdatedArtifacts) {
+              existingUpdatedArtifacts.contents = existingContent;
+            } else {
+              updatedArtifacts.push({
+                name: relativePath,
+                contents: existingContent,
+              });
+            }
+            // If the file is deleted by a previous post-update command, remove the deletion from updatedArtifacts
+            updatedArtifacts = updatedArtifacts.filter(
+              (ua) => ua.name !== '|delete|' || ua.contents !== relativePath
+            );
+          }
+        }
+      }
+
+      for (const relativePath of status.deleted || []) {
+        for (const pattern of fileFilters) {
+          if (minimatch(relativePath, pattern)) {
+            logger.debug(
+              { file: relativePath, pattern },
+              'Post-upgrade file removed'
+            );
+            updatedArtifacts.push({
+              name: '|delete|',
+              contents: relativePath,
+            });
+            // If the file is created or modified by a previous post-update command, remove the modification from updatedArtifacts
+            updatedArtifacts = updatedArtifacts.filter(
+              (ua) => ua.name !== relativePath
+            );
+          }
+        }
+      }
+    }
+  }
+  return { updatedArtifacts, artifactErrors };
+}
+
+export default async function executePostUpgradeCommands(
+  config: BranchConfig
+): Promise<PostUpgradeCommandsExecutionResult | null> {
+  const { allowedPostUpgradeCommands, trustLevel } = getAdminConfig();
+
+  const hasChangedFiles =
+    config.updatedPackageFiles?.length > 0 ||
+    config.updatedArtifacts?.length > 0;
+
+  if (
+    /* Only run post-upgrade tasks if there are changes to package files... */
+    !hasChangedFiles ||
+    trustLevel !== 'high' ||
```

Comment:
```suggestion
```

There was a feature request to not need `trustLevel=high` for post upgrade tasks. So this check will be removed in v25 anyways.

`trustLevel` is used if you like to pass full env to child process, with is not always needed and i do not want to allow high trust on  post upgrade tasks 😉